### PR TITLE
Add codesniffer check

### PR DESCRIPTION
**Pull request by robertpustulka**
Refs https://github.com/cakephp/migrations/pull/326#discussion_r144560089

### CODE REVIEW

**Comment by ADmad** [comment]
@robertpustulka Can you update this to `CakePHP` too please https://github.com/cakephp/migrations/blob/master/.stickler.yml#L3

**Comment by ADmad** [comment]
Still various CS issues need to be fixed https://travis-ci.org/cakephp/migrations/jobs/287581421#L524
Also need to ignore the `comparisons` folder from CS check.

File: phpcs.xml.dist [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<ruleset name="CakePHP Core">
+ <rule ref="./vendor/cakephp/cakephp-codesniffer/CakePHP/ruleset.xml"/>
```
Comment (dereuromark) [inline phpcs.xml.dist]: I hate space indentation. It is soo wrong.
But in this case, would maybe 2 or 4 at least be less wrong than just 1? :)

Comment (robertpustulka) [reply phpcs.xml.dist]: It's a copy-paste from the core repo.

Comment (dereuromark) [reply phpcs.xml.dist]: Part of the problem of space usage. I guess that should also be corrected to what is configured for xml files (the .dist just by addicent didnt get any support here when it was edited, thus it happened).

Comment (robertpustulka) [reply phpcs.xml.dist]: Correct that if you please. 

My doctor told me today not to worry about irrelevant stuff. It's bad for your health :)

File: tests/TestCase/Command/StatusTest.php [hunk 2]
```diff
@@ -133,9 +130,9 @@ public function testExecuteJson()
         $commandTester->execute(['command' => $this->command->getName()] + $params);
         $display = $this->getDisplayFromOutput();
 
-        $expected = '{"status":"down","id":"20150704160200","name":"CreateNumbersTable"},'
-            . '{"status":"down","id":"20150724233100","name":"UpdateNumbersTable"},'
```
Comment (dereuromark) [inline tests/TestCase/Command/StatusTest.php]: I much preferred the first as it is better to read. Does the sniffer actually want this to change?

Comment (robertpustulka) [reply tests/TestCase/Command/StatusTest.php]: Yes. `ERROR | Expected 1 space before ., but 12 found`

Comment (dereuromark) [reply tests/TestCase/Command/StatusTest.php]: Well, the sniffer is wrong here IMO and should be fixed, not the code.

File: phpcs.xml.dist (hunk 1 above)
Comment (dereuromark) [reply phpcs.xml.dist]: haha, then you probably have a complete PR here that should be closed :) Not too valid an argument here IMO.

File: tests/TestCase/Command/StatusTest.php (hunk 2 above)
Comment (robertpustulka) [reply tests/TestCase/Command/StatusTest.php]: I'm not sure. To me the sniffer suggestion looks fine. Actually I'm fine with both.

File: phpcs.xml.dist (hunk 1 above)
Comment (robertpustulka) [reply phpcs.xml.dist]: Alt+Shift+F did some automagic here. Is that OK now?

File: tests/TestCase/Command/StatusTest.php (hunk 2 above)
Comment (dereuromark) [reply tests/TestCase/Command/StatusTest.php]: I opened a ticket https://github.com/cakephp/cakephp-codesniffer/issues/198 for now.

Comment (robertpustulka) [reply tests/TestCase/Command/StatusTest.php]: @dereuromark It's ours it seems:

` 134 | ERROR | Expected 1 space before ., but 12 found (CakePHP.Strings.ConcatenationSpacing.TooManyBefore)
`

**Comment by robertpustulka** [comment]
What's wrong with tests on mysql?

https://travis-ci.org/cakephp/migrations/jobs/288989814

File: tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php [hunk 3]
```diff
@@ -35,4 +35,3 @@ public function down()
             ->update();
     }
 }
-
```
Comment (HavokInspiration) [inline tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: All the modifications to the "comparisons" (this one and below) are going to break the tests as they are probably expected in the comparisons.

Comment (robertpustulka) [reply tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: Sorry, I run the fixer on all folders. I’ll revert those changes.

File: tests/test_app/App/Model/Table/LettersTable.php [hunk 4]
```diff
@@ -9,7 +9,9 @@
  * @link          http://cakephp.org CakePHP(tm) Project
  * @license       http://www.opensource.org/licenses/mit-license.php MIT License
  */
-namespace TestApp\Model\Table;;
+namespace TestApp\Model\Table;
+
+;
```
Comment (HavokInspiration) [inline tests/test_app/App/Model/Table/LettersTable.php]: Unneeded `;`

File: tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php (hunk 3 above)
Comment (HavokInspiration) [reply tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: @robertpustulka This is probably what's currently breaking the MySQL builds.

File: tests/test_app/App/Model/Table/LettersTable.php (hunk 4 above)
Comment (robertpustulka) [reply tests/test_app/App/Model/Table/LettersTable.php]: How about a sniffer for double semicolon?

Comment (dereuromark) [reply tests/test_app/App/Model/Table/LettersTable.php]: Sounds like an improvement ticket for code-sniffer repo :) But out of scope of this PR then.

File: tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php (hunk 3 above)
Comment (ADmad) [reply tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: @HavokInspiration The changes to comparison files is just whitespace change. I doubt that's what causing this error https://travis-ci.org/cakephp/migrations/jobs/288989818#L531

Comment (robertpustulka) [reply tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: Those whitespaces were important it seems :)

Comment (HavokInspiration) [reply tests/comparisons/Diff/addremove/the_diff_add_remove_mysql.php]: @ADmad The real error is here : https://travis-ci.org/cakephp/migrations/jobs/288989818#L558-L565

What you highlighted was just a side-effect of a previous test failing.

**Pull request merged**